python/pyarrow/__init__.py

```diff
@@ -207,6 +207,28 @@ def show_versions():
 
 import pyarrow.types as types
 
+
+if _sys.version_info >= (3, 7):
+    def __getattr__(name):
+        _msg = ("'pyarrow.{0}' is deprecated and will be "
+                "removed in a future version. Use pickle or the pyarrow IPC "
+                "functionality instead.")
+        if name == "SerializationContext":
+            _warnings.warn(_msg.format(name), DeprecationWarning)
```

Comment:
I think we use `FutureWarning` instead of `DeprecationWarning`? (because the latter is silent by default)